transports/webrtc/src/tokio/connection.rs

```diff
@@ -0,0 +1,312 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use futures::stream::FuturesUnordered;
+use futures::{
+    channel::{
+        mpsc,
+        oneshot::{self, Sender},
+    },
+    lock::Mutex as FutMutex,
+    StreamExt,
+    {future::BoxFuture, ready},
+};
+use libp2p_core::muxing::{StreamMuxer, StreamMuxerEvent};
+use webrtc::data::data_channel::DataChannel as DetachedDataChannel;
+use webrtc::data_channel::RTCDataChannel;
+use webrtc::peer_connection::RTCPeerConnection;
+
+use std::task::Waker;
+use std::{
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+use crate::tokio::{error::Error, substream, substream::Substream};
+
+/// Maximum number of unprocessed data channels.
+/// See [`Connection::poll_inbound`].
+const MAX_DATA_CHANNELS_IN_FLIGHT: usize = 10;
+
+/// A WebRTC connection, wrapping [`RTCPeerConnection`] and implementing [`StreamMuxer`] trait.
+pub struct Connection {
+    /// `RTCPeerConnection` to the remote peer.
```

Comment:
```suggestion
    /// [`RTCPeerConnection`] to the remote peer.
```

In case this is ever renamed, we are reminded to update the comment.